go-fuzz-build/main.go

```diff
@@ -737,18 +751,41 @@ func (c *Context) instrumentPackages(blocks *[]CoverBlock, sonar *[]CoverBlock)
 }
 
 func (c *Context) copyDir(dir, newDir string) {
-	c.mkdirAll(newDir)
 	files, err := ioutil.ReadDir(dir)
 	if err != nil {
 		c.failf("failed to scan dir '%v': %v", dir, err)
 	}
+	c.mkdirAll(newDir)
 	for _, f := range files {
+		src := filepath.Join(dir, f.Name())
+		dst := filepath.Join(newDir, f.Name())
 		if f.IsDir() {
-			c.copyDir(filepath.Join(dir, f.Name()), filepath.Join(newDir, f.Name()))
+			c.copyDir(src, dst)
 			continue
 		}
+		c.copyFile(src, dst)
+	}
+}
+
+func (c *Context) copyNotPkgDir(dir, newDir string) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		c.failf("failed to scan dir '%v': %v", dir, err)
+	}
+	for _, f := range files {
+		if strings.HasSuffix(f.Name(), ".go") {
+			// it is Pkg dir
+			return
+		}
+	}
+	c.mkdirAll(newDir)
+	for _, f := range files {
 		src := filepath.Join(dir, f.Name())
 		dst := filepath.Join(newDir, f.Name())
+		if f.IsDir() {
```

Comment:
There's a lot of duplicated code here with copyDir. It is worth unifying them?

Reply:
Done at https://github.com/dvyukov/go-fuzz/pull/311/commits/48e365bef79dd988f812a63fb48aec56d95f84a5